fineract-provider/src/main/java/org/apache/fineract/infrastructure/core/persistence/ExtendedJpaTransactionManager.java

```diff
@@ -29,6 +32,8 @@
 
 public class ExtendedJpaTransactionManager extends JpaTransactionManager {
```

Comment:
Do we really need this? Can't this be accomplished with https://docs.spring.io/spring-framework/docs/current/javadoc-api/org/springframework/transaction/event/TransactionalEventListener.html (aka transaction bound events)? From a first look at this I'd say yes... if you want to keep those custom lifecycle callbacks then you just need to write a transaction event listener and do pretty much the same you do here in the extended transaction manager.